syncer/syncer.go

```diff
@@ -925,11 +960,7 @@ func (s *Syncer) redirectStreamer(pos mysql.Position) error {
 	var err error
 	s.tctx.L().Info("reset global streamer", zap.Stringer("position", pos))
 	s.resetReplicationSyncer()
-	if s.binlogType == RemoteBinlog {
-		s.streamer, err = s.getBinlogStreamer(s.syncer, pos)
-	} else if s.binlogType == LocalBinlog {
-		s.streamer, err = s.getBinlogStreamer(s.localReader, pos)
-	}
+	s.streamer, err = s.streamerProducer.generateStreamer(pos)
```

Comment:
we can remove `getBinlogStreamer`, `getLocalBinlogStreamer` and `getRemoteBinlogStreamer` after we extract the interface.

Reply:
Done